src/modm/container/deque.hpp

```diff
@@ -189,7 +190,7 @@ namespace modm
 		 *
 		 * \todo	check if a simpler implementation is possible
 		 */
-		class const_iterator
+		class const_iterator : public std::iterator<std::bidirectional_iterator_tag, T>
```

Comment:
`std::iterator` has been deprecated in C++17. See [this](https://www.fluentcpp.com/2018/05/08/std-iterator-deprecated/) for more information. Please use the recommended method from the article with type aliases in the iterator class.

Reply:
Didn't it's deprecated. I'm building in C++17 and didn't get any warning...